src/nile/core/account.py

```diff
@@ -79,32 +91,55 @@ async def __init__(self, signer, network, predeployed_info=None, watch_mode=None
             self.address = signer_data["address"]
             self.index = signer_data["index"]
         else:
-            address, index = await self.deploy(watch_mode=watch_mode)
-            self.address = address
-            self.index = index
+            output = await self.deploy(
+                salt=salt, max_fee=max_fee, watch_mode=watch_mode
+            )
+            if output is not None:
```

Comment:
If this `output` is None, `self.address` is never assigned, and the assertion below reverts (is not an invariant because of this). 

This is an issue when a deploy account transaction reverts because instead of showing just the error from the status command, the execution raises an exception from the assertion because self.address is not defined.

I suggest changing the assertion to this to keep the message clear:

```
if hasattr(self, 'address'):
    assert type(self.address) == int
```

To visualize the issue just run `nile setup account` after running a clean node with `nile node`.